README.md

```
@@ -111,6 +116,65 @@ The default configuration
+
+`handle_inject_agent_msg` runs `policy_chain.review()` then `policy_chain.observe()` for every accepted message. `handle_binary_message` runs `policy_chain.review_binary()`. — `hivemind_core/protocol.py:908`, `hivemind_core/protocol.py:457`
+
+**ACL model — whitelist-only, deny-by-default:**
+
+- `allowed_types` is the only ACL field on `HiveMindClientConnection`. There is no message blacklist. — `hivemind_core/protocol.py:92`
+- A freshly created client (via `add-client`) has an **empty** `allowed_types` list and will be denied all messages until the operator explicitly grants types with `allow-msg`. This applies to admin clients too — `Client.is_admin` is informational only and gives no admission bypass.
+
+To grant a message type:
+
+```bash
+$ hivemind-core allow-msg "recognizer_loop:utterance" 1
```

markdownlint-cli2 (0.22.1)

[warning] Dollar signs used before commands without showing output

(MD014, commands-show-output)